source/docs/software/commandbased/builtins.rst

```diff
@@ -0,0 +1,379 @@
+        // Start a flywheel spinning at 50% power
+        () -> m_shooter.shooterSpeed(0.5),
+        // Stop the flywheel at the end of the command
+        () -> m_shooter.shooterSpeed(0.0),
+        // Requires the shooter subsystem
+        m_shooter
+    )
+
+  .. code-tab:: c++
+
+    frc2::StartEndCommand(
+      // Start a flywheel spinning at 50% power
+      [this] { m_shooter.shooterSpeed(0.5); },
+      // Stop the flywheel at the end of the command
+      [this] { m_shooter.shooterSpeed(0.0); },
+      // Requires the shooter subsystem
+      {&m_shooter}
+    )
+
+FunctionalCommand
+^^^^^^^^^^^^^^^^^
+
+``FunctionalCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/FunctionalCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_functional_command.html>`__) accepts four lambdas that constitute the four command lifecycle methods: a ``Runnable``/``std::function<void()>`` for each of ``initialize()`` and ``execute()``, a ``BooleanConsumer``/``std::function<void(bool)>`` for ``end()``, and a ``BooleanSupplier``/``std::function<bool()>`` for ``isFinished()``.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    new FunctionalCommand(
+        // Reset encoders on command start
+        m_robotDrive::resetEncoders,
+        // Start driving forward at the start of the command
+        () -> m_robotDrive.arcadeDrive(kAutoDriveSpeed, 0),
+        // Stop driving at the end of the command
+        interrupted -> m_robotDrive.arcadeDrive(0, 0),
+        // End the command when the robot's driven distance exceeds the desired value
+        () -> m_robotDrive.getAverageEncoderDistance() >= kAutoDriveDistanceInches,
+        // Require the drive subsystem
+        m_robotDrive
+    )
+
+  .. code-tab:: c++
+
+    frc2::FunctionalCommand(
+      // Reset encoders on command start
+      [this] { m_drive.ResetEncoders(); },
+      // Start driving forward at the start of the command
+      [this] { m_drive.ArcadeDrive(ac::kAutoDriveSpeed, 0); },
+      // Stop driving at the end of the command
+      [this] (bool interrupted) { m_drive.ArcadeDrive(0, 0); },
+      // End the command when the robot's driven distance exceeds the desired value
+      [this] { return m_drive.GetAverageEncoderDistance() >= kAutoDriveDistanceInches; },
+      // Requires the drive subsystem
+      {&m_drive}
+    )
+
+PrintCommand
+^^^^^^^^^^^^
+
+``PrintCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/PrintCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_print_command.html>`__) is a subclass of ``InstantCommand`` for printing a string and ending immediately.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    new PrintCommand("This message will be printed!")
+
+  .. code-tab:: c++
+
+    frc2::PrintCommand("This message will be printed!")
+
+
+Dynamically Deciding What Command To Schedule
```

Comment:
Would it be better to rename this title to "Scheduling a Command at Runtime"

Reply:
Not really; this section is more about SelectCommand etc that schedule one of a set of commands based on sensor input etc.